msal-dotnet-articles/acquiring-tokens/desktop-mobile/wam-errors.md

```diff
@@ -0,0 +1,37 @@
+---
+title: Errors associated with Web Account Manager (WAM)
+description: "When building applications using MSAL.NET and the Web Account Manager (WAM), developers may run into issues. This article explains potential errors and mitigations."
+ms.date: 08/30/2023
+---
+
+# Errors associated with Web Account Manager (WAM)
+
+Generally, errors generated by the MSAL component that interacts with the Web Account Manager (WAM) are [automatically converted](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/d25f45f7e356bd92986e412dd68b5d341343aa6a/src/client/Microsoft.Identity.Client/Platforms/Features/RuntimeBroker/WamAdapters.cs#L56) to an instance of <xref:Microsoft.Identity.Client.MsalException>. This enables developers to not worry about the internals and instead handle errors through idiomatic .NET constructs. However, for additional deep dives, developers will often need to investigate specific error messages.
+
+The table below highlights some of the most common errors along with potential mititgation strategies.
+
+>[!WARNING]
+>Error codes and error messages are presented for reference only. We do not recommend manually implementing exception handling strategies based on these and instead use the standard <xref:Microsoft.Identity.Client.MsalException>-based approach.
+
+| Error Code | Retry-able | Error message | Mitigation |
+|:-----------|:-----------|:------------|:-----------|
+| 0          | ❌ | Account credentials missing in WAM, and should be removed. | |
+| 0          | ✅ | Account has previously been signed out of this application. | |
+| 0          | ❌ | Account with id {ID} not found. | |
+| 0          | ❌ | Authentication was blocked because a user switch was detected during authorization type 4. | |
```

Comment:
In some cases, error code == 0 means the error is coming from MSAL C++ internal, not WAM. But we cannot rely on this error code, hence please remove those to reduce confusion. Thanks.